### ISSUE TITLE
Documentation is too sparse

### ISSUE THREAD

**Issue by TheNybbler** [commandlineparser/commandline#415]
The provided documentation only provides very basic examples and does not cover a lot of the different available options.

**Comment by 0x15e** [comment]
I agree with you but fortunately, this is one of the few projects where there are plenty of xmldocs and the source code is relatively easy to read.

That said, there's only so much you can get from that. I can't help but feel like there's a lot this project can do that I'm not able to take advantage of for lack of examples / formal documentation.

**Comment by stefanjarina** [comment]
+1 for this, but I find this a problem of too many .net packages... It is like they teach .net people to not write proper documentation. Coming from Python/Go world it just saddens me as I really love .net core, but am finding it just hard to work with it's ecosystem. Reading a source code and internal api should be a last resort and docs should cover majority of use cases.
Just for comparison check this documentation for click, which is same type of library but for Python: https://click.palletsprojects.com/en/7.x/

**Comment by k4kfh** [comment]
I agree. Once I spent a couple days and sort of poked around, I am finding that this is quite a lovely little library. But the documentation is fragmented, confusing, and often less helpful than just parsing through the source code.

@ericnewton76 I don't know if you're the right person to ask about this, but I saw your name as an author on the wiki. It seems the wiki pages are not publicly editable (or I am just GitHub-illiterate, also a possibility). I think a lot of the sparse documentation could be remedied incrementally if this wiki was publicly editable/PR'able. I'm using this library at work, so I'm having to write documentation within our code about how it works. So I'm already doing the work of writing a lot of documentation (for a pretty complex CLI with verbs, a good example case) but I can't figure out how to contribute the docs I am writing to the wiki on this repo. I suspect I'm not the only one in a situation like this, and I think "crowdsourcing" the documentation may be a good solution if the maintainers have bigger fish to fry at the moment.
